### ISSUE TITLE
MacOS / Apple Silicon fixes (and possibly clang-17 fixes)

### ISSUE THREAD

**Issue by dgarnier** [ascot4fusion/ascot5#122]
To get code to run through tests and tutorials on MacOS.. made some changes mostly to src/Makefile. Also had to make a new abstraction since the structure naming convention on MacOS with clang-17 (which is what I used) has different "mangling". Interestingly, the mangling looks simpler with the newer clang. In general though, ascot2py.py looks different on MacOS arm due to different "long double" and packing. So that file should probably not be kept in the repo.. though I didn't remove it.

I also have a version of this commit against the main branch, but I'm assuming develop with eventually be pulled and this saves a later conflict merge resolution and additional fixes.

**Comment by miekkasarki** [comment]
Yup, hopefully we'll get rid of ascot2py.py someday as we implement the python interface properly. I wasn't aware that there would be an issue with the name mangling but your solution looks clean. So if that works, I'm happy with it. I'll still have to check that all the changes to the Makefile are something that cannot be handled easily with environment variables (I only had a quick glance now).

We currently have a major merge in progress as the GPU branch is being worked, but I'll come back to this once that is done!

**Comment by dgarnier** [comment]
Cool.  Let me know what you are thinking of for getting rid of ascot2py.py.  I've recently moved my code (dipole equilibrium code) over using pybind11 in combination with scikit-build-core for the packaging.  Works nicely.  (Although, I might use "nanobind" instead of pybind11 now that I found out about it and it seems to have a smaller / more direct connection between c++ and python).  In the meantime, I'll rebase my patch so its easier to evaluate.

**Comment by miekkasarki** [comment]
Basically we would pick what we need from ascot2py.py for the Python interface to work, and then maintain those manually instead of automatically creating the whole file. Now that were are able to transfer pointers to input/output data between C and Python directly, it makes sense to have the struct definitions (in Python) in the same module where the corresponding data objects are.

I now merged your changes to develop. The only thing I'm unsure of was that there was this part:

# For MacOS, we need to add the clang library path to the environment
# since built-in clang doesn't have openmp support enabled by default
# ascot2py.py: export CLANG_LIBRARY_PATH=$(shell $(CC) --print-resource-dir)/../..

Since it was commented, I did not include it. However, if this `export CLANG_LIBRARY_PATH=$(shell $(CC) --print-resource-dir)/../..` is in some cases required, we could put some flag to enable it.
